File: src/main/java/kr/hyfata/zero/modules/mailbox/util/MailboxConfigUtil.java
package kr.hyfata.zero.modules.mailbox.util;

import kr.hyfata.zero.ZeroCore;
import kr.hyfata.zero.config.IConfig;
import kr.hyfata.zero.modules.gui.mailbox.MailboxButton;
import kr.hyfata.zero.util.TextFormatUtil;
import org.bukkit.Material;

public class MailboxConfigUtil {
    public static IConfig getConfig() {
        return ZeroCore.configModules.getMailboxConfig();
    }

    public static String getMailBoxTitle() {
        return getConfig().getString("mailbox.title", "ERROR");
    }

    public static int getRows() {
        return getConfig().getConfig().getInt("mailbox.rows", 6);
    }

    public static int[] getRewardItemRowRange() {
        int[] result = new int[2];
        result[0] = getConfig().getConfig().getInt("mailbox.reward_item_row_start", 1);
        result[1] = getConfig().getConfig().getInt("mailbox.reward_item_row_end", 6);
        return result;
    }

    public static Material getErrorMaterial() {
        return Material.getMaterial(getConfig().getString("task.error.item", "RED_CONCRETE"));
    }

    public static int getErrorCustomModelData() {
        return getConfig().getConfig().getInt("task.error.custom_model_data", 0);
    }

    public static Material getSuccessMaterial() {
        return Material.getMaterial(getConfig().getString("task.success.item", "GREEN_CONCRETE"));
    }

    public static int getSuccessCustomModelData() {
        return getConfig().getConfig().getInt("task.success.custom_model_data", 0);
    }

    private static MailboxButton getButton(String path) {
        MailboxButton result = new MailboxButton();
        IConfig config = getConfig();

        result.setPositions(config.getConfig().getIntegerList(path + ".pos"));
        result.setName(TextFormatUtil.getFormattedText(config.getString(path + ".txt", "ERROR")));
        result.setLore(TextFormatUtil.getFormattedText(config.getString(path + ".lore", "")).split("\n"));
        result.setItem(Material.getMaterial(config.getString(path + ".item", "STONE")));
        result.setCustomModelData(config.getConfig().getInt(path + ".custom_model_data", 0));
        return result;
    }

    public static MailboxButton getAllRewardsButton() {
        return getButton("buttons.get_all_rewards");
    }

    public static MailboxButton getPreviousButton() {
        return getButton("buttons.previous");
    }

    public static MailboxButton getNextButton() {
        return getButton("buttons.next");
    }
}

File: src/main/java/kr/hyfata/zero/util/config/MailboxConfigUtil.kt
package kr.hyfata.zero.util.config

import kr.hyfata.zero.ZeroCore
import kr.hyfata.zero.config.AbstractConfig
import kr.hyfata.zero.modules.mailbox.dto.MailboxButton
import kr.hyfata.zero.helper.format.TextFormatHelper
import org.bukkit.Material

object MailboxConfigUtil {
        val config: AbstractConfig
        get() = ZeroCore.Companion.zeroConfig.mailboxConfig

    val mailBoxTitle: String?
        get() = config.getString("mailbox.title", "ERROR")

    val rows: Int
        get() = config.config.getInt("mailbox.rows", 6)

    val rewardItemRowRange: IntArray
        get() {
            val result = IntArray(2)
            config.config.let { result[0] = it.getInt("mailbox.reward_item_row_start", 1) }
            config.config.let { result[1] = it.getInt("mailbox.reward_item_row_end", 6) }
            return result
        }

    val errorMaterial: Material?
        get() = config.getString("task.error.item", "RED_CONCRETE")?.let { Material.getMaterial(it) }

    val errorCustomModelData: Int
        get() = config.config.getInt("task.error.custom_model_data", 0)

    val successMaterial: Material?
        get() = config.getString(
            "task.success.item",
            "GREEN_CONCRETE"
        )?.let {
            Material.getMaterial(
                it
            )
        }

    val successCustomModelData: Int
        get() = config.config.getInt("task.success.custom_model_data", 0)

    private fun getButton(path: String?): MailboxButton {
        val result = MailboxButton()
            

        result.positions = config.config.getIntegerList("$path.pos")
        result.name = config.getString("$path.txt", "ERROR")?.let { TextFormatHelper.getFormattedText(it) }
        config.getString("$path.lore", "")?.let { TextFormatHelper.getFormattedText(it) }?.split("\n".toRegex())
            ?.dropLastWhile { it.isEmpty() }?.let {
                result.setLore(
                    it.toTypedArray()
                )
            }
        result.item = config.getString("$path.item", "STONE")?.let { Material.getMaterial(it) }
        config.config.let { result.customModelData = it.getInt("$path.custom_model_data", 0) }
        return result
    }

    val allRewardsButton: MailboxButton
        get() = getButton("buttons.get_all_rewards")

    val previousButton: MailboxButton
        get() = getButton("buttons.previous")

    val nextButton: MailboxButton
        get() = getButton("buttons.next")
}
